src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -21,16 +21,25 @@
     UniqueTagList getTags();
 
     /**
-     * Returns true if the values inside this object is same as those of the other
+     * Returns true if both persons have the same identity fields (name and telephone).
+     */
+    default boolean isSamePerson(ReadOnlyPerson other) {
+        return other == this || (other != null && other.getName().equals(this.getName())
+                && other.getPhone().equals(this.getPhone()));
+    }
```

Comment:
To add on, indent last two lines a bit more so that `||` doesn't align with `&&` (because they are not at the same level in the condition tree)